### PR TITLE
[FLINK-24635][examples] Clean up flink-examples

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

Clean up flink-examples to remove usage of deprecated APIs. The final commit then adds a flag to fail compilation if a deprecated apis to keep examples up to date going forward. 

## Brief change log

This is a mostly trivial find and replaces of deprecated APIs with their modern counterparts. Each example is its own commit for easy review.

A few, such as AsyncIO, were additionally simplified but nothing was substantially changed. 

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

no

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by sjwiesman** [comment]
@RocMarshal @alibahadirzeybek @NicoK  would any of you like to take a look at this?

File: flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/async/AsyncIOExample.scala [hunk 1]
```diff
@@ -19,53 +19,69 @@
...
+      case _ => throw new IllegalStateException("Unknown mode: " + mode)
     }
-  }
 
-  override def cancel(): Unit = {
-    running = false
+    result.print()
+
+    // execute the program
+    env.execute("Async IO Example")
```
Comment (RocMarshal) [inline flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/async/AsyncIOExample.scala:85]: What about appending the mode information of the async operator into the job name?

**Review by RocMarshal — COMMENTED** [review]
@sjwiesman Thanks for the outstanding contribution.
I left a minor comment.

File: flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/windowing/util/CarSource.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package org.apache.flink.streaming.examples.windowing.util;
```
Comment (RocMarshal) [inline flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/windowing/util/CarSource.java:1]: lack of licenses

File: flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/async/util/SimpleSource.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package org.apache.flink.streaming.examples.async.util;
```
Comment (RocMarshal) [inline flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/async/util/SimpleSource.java:1]: missing of licenses

File: flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/async/AsyncIOExample.scala (hunk 1 above)
Comment (sjwiesman) [reply flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/async/AsyncIOExample.scala:85]: Great idea!

**Pull request closed without merging**